src/architecture/storage.md

```diff
@@ -1,262 +1,158 @@
+family in the
+low-level storage. Keys are represented as byte sequences. Values are
+serialized according to the Protobuf serialization format. All keys are mapped
+to the low-level storage keys in a deterministic manner
+using [object identifiers](#object-identifiers).
+
+### Object Identifiers
+
+Each object in the database has its own unique identifier and metadata. The
+metadata can store various information about the internal state of the object.
+For example, for [`ListIndex`], its length is stored in the metadata.
+Metadata is represented by the [`IndexMetadata`][index-metadata] structure,
+which stores the object identifier and the metadata state.
+
+To obtain object identifiers [object pool][indexes-pool] is used. The pool
+stores identifiers of all available objects. The pool assigns identifiers to
+the objects incrementally as soon as new objects appear. The pool is
+stored in a separate column family. Key in this
+column family is the object identifier, value is the object metadata.
+
+On user level every object is uniquely identified by an address. This address
+maps the object to the identifier from the object pool mentioned before.
+An object address consists of 2 parts:
+
+- **String name,** that may contain uppercase and lowercase Latin letters,
+  digits,
+  underscores `_`, and periods `.`. By convention, object names in services
   should
   start with [the service name][service-name] and a period. For example,
-  the only table in the Cryptocurrency Tutorial is named
+  the only object in the Cryptocurrency Tutorial is named
   `cryptocurrency.wallets`,
   where `cryptocurrency` is the service name, and `wallets` is the own name
-  of the table.
+  of the object.
 - **Optional prefix** presented as a sequence of bytes (`Vec<u8>` in Rust
   terms).
 
-If the prefix is present, the column family identified by the table name
-stores a *group* of tables, rather than a single table.
-In this case, prefixes are used to distinguish tables within the group.
+An object key in RocksDB is a 32-elements byte array. The first 16 bytes are an
+object identifier
+obtained using the object pool. The second 16 bytes are the key of an item
+stored inside the object. In the database, it looks like this:
+`(identifier | key)` - 0x00000000000000100000000000000002
 
 !!! note "Example"
-    Key `key` at the table with name `exchange.crypto` and prefix `BTC`
-    (`0x42 0x54 0x43` in ASCII) matches key
-    `0x42 0x54 0x43 | key` in the column family in RocksDB named
-    `exchange.crypto`.
-
-!!! warning
-    It is strongly advised not to admit
-    a situation when a table prefix in a table group starts with
-    another table prefix in the same group. Such cases may cause unpredictable
-    collisions between logically different keys and elements.
-    As a possible way to avoid this, prefixes within the group may have
-    a fixed byte size.
+    Suppose we have a list with the address `(" exchange.crypto "," BTC ")` in
```

Comment:
The paragraph above says that 

> Key in this column family [object pool] is the object identifier

Then how are names mapped to object identifiers?

Reply:
edited